### PR TITLE
Lab8

### LINKED ISSUE #618: Lab 8 - Project Layout

### PR DESCRIPTION
**Pull request by nicko-lee**
Fixes #618 

Review of colleague's PR #555 

#### Changes proposed in this PR:

Implement lab 6 + restructure packages

### CODE REVIEW

**Review by kasunfdo — APPROVED** [review]

**Comment by kasunfdo** [comment]
LGTM

File: 08_project/nickolee/cmd/puppy-server/main.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	store "github.com/anz-bank/go-course/08_project/nickolee/pkg/puppy/store"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	// instantiate new puppy store
+	store := store.NewMapStore()
+	// create a puppy and store in store (no pun intended lol)
+	createdPuppy, _ := store.CreatePuppy(&puppy.Puppy{Breed: "The Hound", Colour: "Of Baskerville", Value: 12300.90})
```
Comment (anzboi) [inline 08_project/nickolee/cmd/puppy-server/main.go:18]: Get in habbit of not ignoring the errors
In main, this typically looks like
```go
var, err := store.CreatePuppy(...)
if err != nil {
    panic(err)
    // log.Fatal(err) // typical in servers
}
```

**Review by anzboi — CHANGES_REQUESTED** [review]
Almost there,
Just want to get in the habbit of checking errors judiciously. You can ignore them if you are ABSOLUTELY sure there cannot be an error, but this is rare and should not be done lightly.

File: 08_project/nickolee/cmd/puppy-server/main.go (hunk 1 above)
Comment (nicko-lee) [reply 08_project/nickolee/cmd/puppy-server/main.go:18]: Thanks for the tip. Will definitely keep that in mind going forward. 

However for the purposes of this lab I couldn't figure out how to hit 100% coverage in my main_test.go so I won't be able to pass CI :sob:

Comment (anzboi) [reply 08_project/nickolee/cmd/puppy-server/main.go:18]: Fair enough

**Review by anzboi — APPROVED** [review]
LGTM

**Pull request merged**